src/Bicep.Cli/Logging/BicepDiagnosticLogger.cs

```diff
@@ -33,10 +35,25 @@ public void LogDiagnostic(Uri fileUri, IDiagnostic diagnostic, ImmutableArray<in
             this.logger.Log(ToLogLevel(diagnostic.Level), message);
 
             this.HasLoggedErrors |= diagnostic.Level == DiagnosticLevel.Error;
+
+            // Increment counters
+            if (diagnostic.Level == DiagnosticLevel.Warning) { this.WarningCount++; }
+            if (diagnostic.Level == DiagnosticLevel.Error) { this.ErrorCount++; }
+        }
+
+        public void LogSummary() 
+        {
+            var summary = $"Build {(this.HasLoggedErrors ? "failed" : "succeeded")}: {this.WarningCount} Warning(s), {this.ErrorCount} Error(s)";
+            
+            this.logger.Log(ToLogLevel(DiagnosticLevel.Info), summary);
         }
 
         public bool HasLoggedErrors { get; private set; }
```

Comment:
>HasLoggedErrors [](http://example.com/codeflow?start=20&length=15)

nit - `HasLoggedErrors` can now be calculated from `ErrorCount`, but it's not a big deal to keep as-is.

Reply:
good spot, I'll update.